go/test/endtoend/vtgate/reservedconn/get_lock_test.go

```diff
@@ -255,3 +255,19 @@ func TestLocksResetLockOnTimeout(t *testing.T) {
 	utils.AssertContainsError(t, conn, `select is_free_lock('lock')`, "held locks released")
 	utils.AssertMatches(t, conn, `select is_free_lock('lock')`, `[[INT64(1)]]`)
 }
+
+func TestLockWaitOnConnTimeoutWithTxNext(t *testing.T) {
```

Comment:
all the `time.Sleep`s make me nervous that we are introducing flakiness, but I have not alternative to offer

Reply:
this is to ensure that timeouts happen